### PR TITLE
Allow WorkingDirectory to be specified in [Service]

### PR DESCRIPTION
**Pull request by traylenator**
#### Pull Request (PR) description

Support use of `WorkingDirectory` directive in `[Service]` section of `systemd::manage_unit` and `systemd::manage_dropin`

https://www.freedesktop.org/software/systemd/man/systemd.exec.html#WorkingDirectory=

#### This Pull Request (PR) fixes the following issues

Fixes: #299

### CODE REVIEW

File: REFERENCE.md [hunk 1]
```diff
@@ -2316,6 +2316,7 @@ Struct[{
     Optional['AmbientCapabilities']       => Variant[Pattern[/^CAP_[A-Z_]+$/],Array[Pattern[/^CAP_[A-Z_]+$/],1]],
     Optional['User']                      => String[1],
     Optional['Group']                     => String[1],
+    Optional['WorkingDirectory']          => String,
```
Comment (smortex) [inline REFERENCE.md:2319]: Without an explicit minimum length, I trend to think that it was forgotten.  So I suggest:

```suggestion
    Optional['WorkingDirectory']          => String[0],
```

File: spec/type_aliases/systemd_unit_service_spec.rb [hunk 2]
```diff
@@ -39,4 +39,10 @@
   it { is_expected.to allow_value({ 'EnvironmentFile' => ['/etc/sysconfig/foo', '-/etc/sysconfig/foo-bar'] }) }
   it { is_expected.not_to allow_value({ 'EnvironmentFile' => '-/' }) }
   it { is_expected.not_to allow_value({ 'EnvironmentFile' => 'relative-path.sh' }) }
+
+  it { is_expected.to allow_value({ 'WorkingDirectory' => '/var/lib/here' }) }
+  it { is_expected.to allow_value({ 'WorkingDirectory' => '-/var/lib/here' }) }
+  it { is_expected.to allow_value({ 'WorkingDirectory' => '~' }) }
+  it { is_expected.to allow_value({ 'WorkingDirectory' => '' }) }
+
```
Comment (smortex) [inline spec/type_aliases/systemd_unit_service_spec.rb:47]: Removing this blank line should fix CI.

**Review by smortex — APPROVED** [review]
Looks good, fixing CI should be straightforward, and while here maybe we can put an explicit 0-sized string as minimum length.

File: REFERENCE.md (hunk 1 above)
Comment (traylenator) [reply REFERENCE.md:2319]: You need to allow `WorkingDirectory=''` to allow people to set back to the default.
Folk could explicitly set back to the default I guess but other similar directives support `''`.

Comment (traylenator) [reply REFERENCE.md:2319]: Sorry misinterpreted - Explicit 0 ?  that's the default anyway surely?

**Comment by smortex** [comment]
> ```
> Failure/Error: configure_beaker
> RuntimeError:
>   No repository installation step for archlinux yet...
> ```
> 
> So that's unrelated, but something we should fix.

I opened this today: https://github.com/voxpupuli/beaker_puppet_helpers/pull/31 :wink:

**Pull request merged**